KubeArmor/BPF/tests/checksyscall.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Authors of KubeArmor
+
+realpath() {
+    CURR=$PWD
+
+    cd "$(dirname "$0")"
+    LINK=$(readlink "$(basename "$0")")
+
+    while [ "$LINK" ]; do
+        cd "$(dirname "$LINK")"
+        LINK=$(readlink "$(basename "$1")")
+    done
+
+    REALPATH="$PWD/$(basename "$1")"
+    echo "$REALPATH"
+
+    cd $CURR
+}
+
+BPF_HOME=`dirname $(realpath "$0")`/..
+
+if command -v sudo &> /dev/null
+then
+    SUDO="sudo"
+fi
+
+SYSCALL=${1#*/}
+
+if ! $SUDO syscheck $1 ; then
+    echo "Disabeling $SYSCALL ..."
+    sed -i "s#// $SYSCALL start#/*#" $BPF_HOME/system_monitor.c
+    sed -i "s#// $SYSCALL end#*/#"  $BPF_HOME/system_monitor.c
+    echo "$SYSCALL" >> ignore.lst
+fi
```

Comment:
Can we use (if preprocessor)[https://gcc.gnu.org/onlinedocs/cpp/If.html] for conditional compilation instead of brute removing contents?

Reply:
That's how things are in the Kernel as well so...
Ref https://elixir.bootlin.com/linux/latest/source/include/linux/security.h#L1769